### PR TITLE
Display next batch of transactions on scroll

### PR DESCRIPTION
**Pull request by arpadt**
https://bullioncapital.atlassian.net/browse/ABX-4708

Removed grey space from the bottom of the page
Created new action and epic to load next page
Modified reducers to handle payload from both next page and polling
Created new function to fetch next page

### CODE REVIEW

File: src/containers/Routing.tsx [hunk 1]
```diff
@@ -24,13 +24,13 @@ const mapStateToProps = (state: RootState) => ({
 type Props = RouteComponentProps<any> & ReturnType<typeof mapStateToProps>
 const RoutingPresentation: React.SFC<Props> = props =>
   props.isLoggedIn ? (
-    <div className="columns is-marginless is-centered">
+    <div className="columns is-marginless is-centered is-marginless">
```
Comment (desnor) [inline src/containers/Routing.tsx]: oops, this is duplicated

File: src/store/actions/transactions.ts [hunk 2]
```diff
@@ -14,6 +14,9 @@ export const accountTransactionsLoaded = createStandardAction('ACCOUNT_TRANSACTI
 >()
 export const loadAccountTransactions = createStandardAction('LOAD_ACCOUNT_TRANSACTIONS')<string>()
 export const loadNextTransactionPage = createAction('LOAD_NEXT_TRANSACTION_PAGE')
+export const nextTransactionPageLoaded = createStandardAction('NEW_TRANSACTION_PAGE_LOADED')<
```
Comment (desnor) [inline src/store/actions/transactions.ts]: should this be `NEXT_...` ?

File: src/store/reducers/transactions.ts [hunk 3]
```diff
@@ -41,9 +56,17 @@ function transactionOperations(
 ): TransactionOperationView[] {
   switch (action.type) {
     case getType(loadAccountTransactions):
+    case getType(setActiveAccount):
       return []
     case getType(accountTransactionsLoaded):
       return action.payload.operations
+        .filter(({ id }) => state.every(({ id: existingId }) => id !== existingId))
```
Comment (desnor) [inline src/store/reducers/transactions.ts]: I think it'd be nice to put this logic in a separate function outside this case statement, to help remove the clutter and repetition, making it easier to grasp what's going on in a quick look. eg.

```typescript
uniqueIdInList = (list) => ({ id }) => list.every(({ id: existingId }) => id !== existingId)
```
which could then be called with something like this
```typescript
case getType(...):
  return action.payload.operations
    .filter(uniqueIdInList(state))
    .concat(state)
```

File: src/store/reducers/transactions.ts [hunk 4]
```diff
@@ -41,9 +56,17 @@ function transactionOperations(
...
+        .concat(state)
+    case getType(nextTransactionPageLoaded):
+      const newTransactions = action.payload.operations.filter(({ id }) =>
+        state.every(({ id: existingId }) => id !== existingId),
+      )
+      return state.concat(newTransactions)
```
Comment (desnor) [inline src/store/reducers/transactions.ts]: and here like this

```typescript
case getType(...):
  return state.concat(
    action.payload.operations.filter(uniqueIdInList(state))
  )
```

File: src/store/reducers/transactions.ts (hunk 3 above)
Comment (arpadt) [reply src/store/reducers/transactions.ts]: Yes, I contemplated to create a function but decided not to pollute the reducer file.
done

File: src/store/actions/transactions.ts (hunk 2 above)
Comment (arpadt) [reply src/store/actions/transactions.ts]: oops
fixed it

File: src/containers/Routing.tsx (hunk 1 above)
Comment (arpadt) [reply src/containers/Routing.tsx]: rather twice than not at all
fixed it

**Pull request merged**